keps/sig-node/5573-cgroup-v1-unsupported/README.md

```diff
@@ -0,0 +1,565 @@
+
+5. **Community alignment**: Provide clear signals to the Kubernetes community about the deprecation timeline and encourage adoption of cgroup v2.
+
+6. **Remove cgroup v1 code**: Removing the code is the next logic step once we stop testing it. This will clean up
+the codebase.
+
+### Non-Goals
+
+NA.
+
+## Proposal
+
+This proposal builds upon the foundation laid by KEP-4569 (Moving cgroup v1 support into maintenance mode) and formally removes cgroup v1 code from kubelet.
+
+### Risks and Mitigations
+
+The primary risks involve potential disruptions for users who have not yet migrated to cgroup v2:
+
+1. **Existing clusters running cgroup v1**: Users running Kubernetes on hosts with cgroup v1 will need to either:
+   - Migrate their hosts to cgroup v2 (recommended)
+   - Explicitly set `FailCgroupV1=false` to continue using cgroup v1 (not recommended)
+
+2. **Workload compatibility**: Users depending on technologies that require specific versions for cgroup v2 support:
+   - OpenJDK / HotSpot: jdk8u372, 11.0.16, 15 and later
+   - NodeJs 20.3.0 or later
+   - IBM Semeru Runtimes: jdk8u345-b01, 11.0.16.0, 17.0.4.0, 18.0.2.0 and later
+   - IBM SDK Java Technology Edition Version (IBM Java): 8.0.7.15 and later
+   - Third-party monitoring and security agents need to support cgroup v2
+
+**Mitigations**:
+- Provide comprehensive migration documentation and guidance
+- Change the KubeletConfig field FailCgroupV1 to false.
+- Clear warning messages when cgroup v1 is detected
+- Community support through migration period
+- Advance notice through multiple release cycles
+
+## Design Details
+
+This enhancement primarily involves configuration changes and messaging updates, building on the infrastructure already implemented in KEP-4569.
+
+### Move to Deprecated as first phase
+
+### Enable FailCgroupV1 by default
+
+The key technical change is to modify the default value of the kubelet config api `FailCgroupV1` from `false` to `true`. This change will be implemented in the kubelet configuration types.
+
+Current behavior:
+```go
+// Default: false (cgroup v1 support enabled by default)
+FailCgroupV1: false,
+```
+
+Proposed behavior:
+```go
+// Default: true (cgroup v1 support disabled by default)
+FailCgroupV1: true,
+```
+
+### Update warning messages and events
+
+Update the warning messages and events introduced in KEP-4569 to reflect the new unsupported status:
+
+From (maintenance mode):
+```golang
+klog.Warning("cgroup v1 detected. cgroup v1 support has been transitioned into maintenance mode, please plan for the migration towards cgroup v2. More information at https://git.k8s.io/enhancements/keps/sig-node/4569-cgroup-v1-maintenance-mode")
+```
+
+To (deprecated):
+```golang
+klog.Warning("cgroup v1 detected. cgroup v1 support is deprecated and will be removed in a future release. Please migrate to cgroup v2. More information at https://git.k8s.io/enhancements/keps/sig-node/5573-cgroup-v1-unsupported")
```

Comment:
We use the term `deprecated` here while using `unsupported` throughout the document so far. Can we use a single term? Maybe prefer `deprecated` as that's what we used during the call today.